scripts/build.js

```diff
@@ -84,6 +90,40 @@ function build(options) {
 }
 
 /**
+ * Resolve node-gyp command to invoke
+ *
+ * @api private
+ */
+function resolveNodeGyp() {
+  if (process.jsEngine === 'chakracore') {
+    // For node-chakracore, check if node-gyp is in the path.
+    // If yes, use it instead of using node-gyp directly from
+    // node_modules because the one in node_modules is not
+    // compatible with node-chakracore.
+    var nodePath = path.dirname(process.execPath);
+    var nodeGypName = process.platform === 'win32' ? 'node-gyp.cmd' : 'node-gyp';
+    var globalNodeGypBin = process.env.Path.split(';').filter(function(envPath) {
```

Comment:
We have node-gyp as a dependency in our package.json to get around these exact issues. You could bump that dependency to the minimum safe version.